internal/mode/static/usage/job_worker_test.go

```diff
@@ -34,64 +36,145 @@ func TestCreateUsageJobWorker(t *testing.T) {
 		},
 	}
 
-	ngfPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: "nginx-gateway",
-			Name:      "ngf-pod",
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					Kind: "ReplicaSet",
-					Name: "ngf-replicaset",
+	tests := []struct {
+		listCalls func(_ context.Context, object client.ObjectList, _ ...client.ListOption) error
+		getCalls  func(_ context.Context, _ types.NamespacedName, object client.Object, _ ...client.GetOption) error
+		expErr    error
+		name      string
+		expData   usage.ClusterDetails
+	}{
+		{
+			name: "succeeds",
+			listCalls: func(_ context.Context, object client.ObjectList, _ ...client.ListOption) error {
+				switch typedList := object.(type) {
+				case *v1.NodeList:
+					typedList.Items = append(typedList.Items, v1.Node{})
+					return nil
+				case *appsv1.ReplicaSetList:
+					typedList.Items = append(typedList.Items, *ngfReplicaSet)
+					return nil
+				}
+				return nil
+			},
+			getCalls: func(_ context.Context, _ types.NamespacedName, object client.Object, _ ...client.GetOption) error {
+				switch typedObject := object.(type) {
+				case *v1.Namespace:
+					typedObject.Name = metav1.NamespaceSystem
+					typedObject.UID = "1234abcd"
+					return nil
+				}
+				return nil
+			},
+			expData: usage.ClusterDetails{
+				Metadata: usage.Metadata{
+					UID:         "1234abcd",
+					DisplayName: "my-cluster",
+				},
+				NodeCount: 1,
+				PodDetails: usage.PodDetails{
+					CurrentPodCounts: usage.CurrentPodsCount{
+						PodCount: 1,
+					},
 				},
 			},
+			expErr: nil,
 		},
-	}
-
-	kubeSystem := &v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: metav1.NamespaceSystem,
-			UID:  "1234abcd",
-		},
-	}
-
-	k8sClient := fake.NewFakeClient(&v1.Node{}, ngfReplicaSet, ngfPod, kubeSystem)
-	reporter := &usagefakes.FakeReporter{}
-
-	worker := usage.CreateUsageJobWorker(
-		zap.New(),
-		k8sClient,
-		reporter,
-		config.Config{
-			GatewayPodConfig: config.GatewayPodConfig{
-				Namespace: "nginx-gateway",
-				Name:      "ngf-pod",
+		{
+			name: "collect node count fails",
+			listCalls: func(_ context.Context, object client.ObjectList, _ ...client.ListOption) error {
+				switch object.(type) {
+				case *v1.NodeList:
+					return errors.New("failed to collect node list")
+				}
+				return nil
 			},
-			UsageReportConfig: &config.UsageReportConfig{
-				ClusterDisplayName: "my-cluster",
+			getCalls: func(_ context.Context, _ types.NamespacedName, _ client.Object, _ ...client.GetOption) error {
+				return nil
 			},
+			expData: usage.ClusterDetails{},
+			expErr:  errors.New("failed to collect node list"),
```

Comment:
do we really need to define an error? When we check we do this:
```
			if test.expErr != nil {
				g.Expect(reporter.ReportCallCount()).To(Equal(0))
			} else {
				_, data := reporter.ReportArgsForCall(0)
				g.Expect(data).To(Equal(test.expData))
			}
```

bool should be enough